back/spec/acceptance/static_pages_spec.rb

```diff
@@ -101,6 +103,72 @@
         expect(json_response.dig(:data, :attributes, :code)).to eq 'custom'
       end
 
+      describe 'updating topics' do
+        let(:projects_filter_type) { 'topics' }
+        let(:topic1) { create(:topic) }
+        let(:topic2) { create(:topic) }
+        let(:topic_ids) { [topic1.id, topic2.id] }
+
+        example_request 'set topics for projects list' do
+          json_response = json_parse(response_body)
+          expect(response_status).to eq 200
+          expect(json_response.dig(:data, :relationships, :topics, :data).length).to eq(2)
+        end
+
+        context 'when wrong project filter type is set' do
+          let(:projects_filter_type) { 'areas' }
+
+          example_request 'attempt to update page' do
+            expect(response_status).to eq 422
+          end
+        end
+
+        context 'when no topic ids in request' do
+          let(:topic_ids) { [] }
```

Comment:
That's good how it is now, but maybe the next time it makes sense to better add such a spec to model level. Because acceptance specs are much slower than unit specs.